### PR TITLE
HPCC-20631 Set session id context

### PR DESCRIPTION
**Pull request by mayx**
- Set the session id context of an ssl context to be "hpccsystems", a
  value unique to our applicaiton.

Signed-off-by: mayx <yanrui.ma@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by mayx** [comment]
@mckellyln please review.

File: system/security/securesocket/securesocket.cpp [hunk 1]
```diff
@@ -896,6 +900,9 @@ class CSecureSocketContext : implements ISecureSocketContext, public CInterface
         {
             throw MakeStringException(-1, "ctx can't be created");
         }
+
+        setSessionIdContext();
```
Comment (mckellyln) [inline system/security/securesocket/securesocket.cpp]: I think setting the ctx session is meaningful only on the server side ?
If that is true, should we do this only for server, ie:
```
        if(sockettype == ServerSocket)
            setSessionIdContext();
```

**Comment by mckellyln** [comment]
@mayx a few questions:
1). does it make sense to set on client side ?
2). would it make sense to have different ids for different components or use cases ?
3). what is the point of setting the id, are you planning on adding session re-use/cache support ?

**Comment by mayx** [comment]
1) it's only needed on the server side
2) Not for our use case
3) The purpose is to fix an error that happens with SSL_accept when client certificate verification is enabled and the client sends in a certificate. Client certificate verification is usually not enabled.

**Comment by mckellyln** [comment]
@mayx ok, thanks - so should it just be done before SSL_accept() or only if sockettype == ServerSocket ?
Other than that - looks good.

**Comment by mayx** [comment]
Good point. The man page states it's only "useful" for server side and it doesn't seem to affect client side functioning. But to stay on the safe side, I'll add the check of sockettype.

**Comment by mayx** [comment]
@mckellyln change uploaded please review again.

**Comment by mckellyln** [comment]
@mayx looks good.

**Review by mckellyln — APPROVED** [review]
@mayx approved

**Pull request merged**